backend/docker.go

```diff
@@ -343,14 +347,19 @@ func (p *dockerProvider) Start(ctx gocontext.Context, startAttributes *StartAttr
 	logger.WithFields(logrus.Fields{
 		"config":      fmt.Sprintf("%#v", dockerConfig),
 		"host_config": fmt.Sprintf("%#v", dockerHostConfig),
-	}).Debug("starting container")
+	}).Debug("creating container")
 
+	// FIXME: This doesn't seem to create the container with the Config and HostConfig
 	container, err := p.client.CreateContainer(docker.CreateContainerOptions{
 		Config:     dockerConfig,
 		HostConfig: dockerHostConfig,
 	})
+	container.Config = dockerConfig
+	container.HostConfig = dockerHostConfig
```

Comment:
Why do we have to do this and the override below?

Reply:
I don't know! But I found that the containers would actually get the CPU sets they were supposed to only when these two lines were present. (I saw similar behavior when looking up requeued jobs [here](https://github.com/travis-ci/worker/blob/e18395ce242e48a54ca7549d018fc3a34f7fc19a/backend/docker.go#L389-L390).)

It really makes no sense, though, so I have doubts.

Reply:
Arrrrrrrgh.  I'm doubly argh about this because of my current effort to switch docker client libraries.  I'll deal with it when the time comes...